### PR TITLE
Install, configure, initialise Cloudwatch-agent during Vault node provisioning

### PR DESCRIPTION
**Pull request by repping**
This code installs, configures and initialises the AWS Cloudwatch-agent on Vault nodes during provisioning.

The following metrics are monitored and send to AWS Cloudwatch:
 - memory_used (precentage)
 - disk_usage (percentage) for mount points "/" and "/opt/vault"

Note 1: A dashboard is will not be created, so currently metrics can only be graphed by manually selecting/filtering them in the Cloudwatch webgui.
Note 2: Terraform fmt -recursive caused some extra whitespace changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+examples/cloudwatch/init.md
+examples/cloudwatch/terraform.tfvars
+examples/cloudwatch/README.md
```
Comment (robertdebock) [inline .gitignore:3]: README.md is good to push.

Comment (robertdebock) [reply .gitignore:3]: Besides that, please place the `.gitignore` file in the `examples/cloudwatch` directory.

File: examples/cloudwatch/main.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+# Call the module.
+module "vault" {
+  certificate_arn  = aws_acm_certificate.default.arn
+  name             = var.name
+  source           = "../../"
+  key_filename     = "id_rsa.pub"
+  size             = "development"
+  allow_ssh        = true
```
Comment (robertdebock) [inline examples/cloudwatch/main.tf:31]: I would try to remove as many variables default.

File: examples/cloudwatch/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+output "vault_url" {
+  description = "The URL to this Vault installation."
+  value       = "https://${cloudflare_record.default.hostname}:8200/ui/"
```
Comment (robertdebock) [inline examples/cloudwatch/outputs.tf:3]: I'll rewrite this later to route53. Not a blocker for approval.

File: examples/cloudwatch/providers.tf [hunk 4]
```diff
@@ -0,0 +1,6 @@
+provider "aws" {
+  region = "eu-north-1"
+}
+provider "cloudflare" {
```
Comment (robertdebock) [inline examples/cloudwatch/providers.tf:4]: I'll remove this later, when switched to Route53.

File: examples/cloudwatch/variables.tf [hunk 5]
```diff
@@ -0,0 +1,36 @@
+variable "name" {
+  description = "The name of the vault cluster in 3 to 5 characters. Changes in runtime would re-deploy a new cluster, data from the old cluster would be lost."
+  type        = string
+  default     = "unset"
+  validation {
+    condition     = length(var.name) >= 3 && length(var.name) <= 5 && var.name != "default"
+    error_message = "Please use a minimum of 3 and a maximum of 5 characters. \"default\" can't be used because it is reserved."
+  }
+}
+variable "cloudflare_token" {
```
Comment (robertdebock) [inline examples/cloudwatch/variables.tf:10]: I'll remove this later. (Route53)

File: examples/cloudwatch/variables.tf [hunk 6]
```diff
@@ -0,0 +1,36 @@
+variable "name" {
```
Comment (robertdebock) [inline examples/cloudwatch/variables.tf:1]: The module itself also has a variable "name", which defaults to `unset`. Not sure if this is required here.

File: examples/cloudwatch/versions.tf [hunk 7]
```diff
@@ -0,0 +1,12 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.27.0"
+    }
+    cloudflare = {
```
Comment (robertdebock) [inline examples/cloudwatch/versions.tf:7]: I'll remove later.

File: user_data_vault.sh.tpl [hunk 8]
```diff
@@ -17,6 +17,52 @@ if [ "${audit_device}" = "true" ] ; then
...
+                        ]
+                ],
+                "metrics_collected": {
+                        "disk": {
+                                "measurement": [
+                                        "used_percent"
+                                ],
+                                "metrics_collection_interval": 60,
+                                "resources": [
+                                        "/","/opt/vault"
```
Comment (robertdebock) [inline user_data_vault.sh.tpl:43]: I'm not 100% sure; if this variable is mentioned in `variables.tf`, please use the `var.SOMETHING` here.

**Review by robertdebock — CHANGES_REQUESTED** [review]
Looks good, a few remarks, some may be incorrect from my side, in that case, just mark them as resolved.

**Comment by repping** [comment]
Resolved all suggestions.

File: examples/cloudwatch/main.tf [hunk 9]
```diff
@@ -0,0 +1,42 @@
+# Make a certificate.
+resource "aws_acm_certificate" "default" {
+  domain_name = "default.${var.domain}"
```
Comment (robertdebock) [inline examples/cloudwatch/main.tf:3]: The domain name `default` is already used in the `default` scenario. Please modify this to `watch` or some other unique value.

**Review by robertdebock — CHANGES_REQUESTED** [review]

File: examples/cloudwatch/main.tf (hunk 9 above)
Comment (repping) [reply examples/cloudwatch/main.tf]: I have created a variable for it and set the default value to "watch".
This variable is used multiple times, including for the Vault deployment name. (see vault module variable "name")
Does it seem quirky to you to use "var.subdomain" here? (see main.tf of the example)

**Review by robertdebock — APPROVED** [review]
Looks good. I'll merge!

**Pull request merged**